rfcs/text/0000-rfc-resource-tagging.md

```diff
@@ -0,0 +1,200 @@
+
+### User Stories
+
+* As a DCP developer, I would like to see how much a particular component of my service is costing in a particular test
+  environment, such as a routine scale test.
+* As a DCP operator, I would like to keep an eye on the running costs incurred by DCP services, to identify unexpected
+  cost increases or service availability issues.
+* As a DCP security lead, I would like to identify resource provisioning actions performed in the production account and
+  track them by operator identity and service.
+
+## Detailed Design
+
+We will use
+[AWS cost allocation tags](https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/custom-tags.html#allocation-how)
+and [GCP labels](https://cloud.google.com/resource-manager/docs/creating-managing-labels) to tag all resources that can
+be tagged. For resources that support tagging/labeling and constitute a significant portion of the IaaS budget (over 1%),
+tagging will be required. For other resources that support tagging/labeling, it is encouraged, but not required.
+
+#### AWS Services that Require Resource Tags
+
+AWS services that specifically require tagging because they have been previously identified as cost drivers are:
+
+* AWS Lambda
+* API Gateway
+* S3
+* EC2 (including all subcomponents that are cost drivers)
+* RDS
+* Elasticsearch Service
+* ELB
+* ECS
+* DynamoDB
+ 
+#### Standard Cost Control Tag/Label Keys
+
+The following tags (AWS) or labels (GCP) **MUST** be set on all HCA DCP assets for which the following conditions apply:
+
+* Their underlying services support tagging/labeling
+* Their underlying services amount to 1% or greater cumulative IaaS spend in the HCA DCP budget in the prior calendar month
+
+   **NOTE** : keys and values are case-sensitive
+
+Name        | Description
+------------|-------------------------------
+`project`   | The value for the project key shall be `dcp`
+`env`       | The value for the env key shall be the stage that the resource applies to, this can be `{dev,staging,integration,prod}`
+`owner`     | The email of a team or individual responsible for operating the asset, for example: dss-team@data.humancellatlas.org
+`service`   | The name of the HCA DCP service that manages the resource, for example: DSS, IngestService
+`Name`      | A human-readable name for the asset. This can be a custom name or a combination of other tags, such as `dss-notify-dev` (service-subcomponent-env). The name should identify the service subcomponent that the asset belongs to, if any. (The value of this tag shows up as the asset name in the AWS Console UI.)
+`managedBy` | The name of a deployment management tool managing this asset, for example: `terraform`, `cloudformation`, `chalice`
+
+
+#### Termination policy
+Assets that are not tagged in accordance with the above are subject to termination.
+
+* After this RFC takes effect, a 90 day grace period will apply, in which all DCP services are expected to implement tagging.
```

Comment:
When does an RFC take effect? When the review process concludes or at some later time decided by the POs?

Reply:
Per the [RFC Process](https://github.com/HumanCellAtlas/dcp-community/blob/master/rfcs/text/0001-rfc-process.md#approving-rfcs):
>NOTE: The existence of an approved software RFC does not indicate any particular priority or commitment to implement the RFC, nor is an approved RFC a green light to implement. Approved RFCs simply demonstrate community agreement on a technical decision. Product priorities are managed and implementations scheduled via the DCP PM planning process.